src/overlays/actors/ovl_Bg_Spot03_Taki/z_bg_spot03_taki.c

```diff
@@ -27,15 +30,142 @@ const ActorInit Bg_Spot03_Taki_InitVars = {
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->bufferIndex = 0;
+    this->openingAlpha = 255.0f;
+    BgSpot03Taki_ApplyOpeningAlpha(this, 0);
+    BgSpot03Taki_ApplyOpeningAlpha(this, 1);
+    this->actionFunc = func_808ADEF0;
+}
+
+void BgSpot03Taki_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot03Taki* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void func_808ADEF0(BgSpot03Taki* this, GlobalContext* globalCtx) {
+    if (this->state == WATERFALL_CLOSED) {
+        if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+            this->state = WATERFALL_OPENING_ANIMATED;
+            this->timer = 40;
+            func_800800F8(globalCtx, 0x1004, -0x63, NULL, 0);
+        }
+    } else if (this->state == WATERFALL_OPENING_IDLE) {
+        this->timer--;
+        if (this->timer < 0) {
+            this->state = WATERFALL_OPENING_ANIMATED;
+        }
+    } else if (this->state == WATERFALL_OPENING_ANIMATED) {
+        if (this->openingAlpha > 0) {
+            this->openingAlpha -= 5;
+            if (this->openingAlpha <= 0.0f) {
+                func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+                this->timer = 400;
+                this->state = WATERFALL_OPENED;
+                this->openingAlpha = 0;
+            }
+        }
+    } else if (this->state == WATERFALL_OPENED) {
+        this->timer--;
+        if (this->timer < 0) {
+            this->state = WATERFALL_CLOSING;
+        }
+    } else if (this->state == WATERFALL_CLOSING) {
+        if (this->openingAlpha < 255.0f) {
+            this->openingAlpha += 5.0f;
+            if (this->openingAlpha >= 255.0f) {
+                func_8003EC50(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+                this->state = WATERFALL_CLOSED;
+                this->openingAlpha = 255.0f;
+                Flags_UnsetSwitch(globalCtx, this->switchFlag);
+            }
+        }
+    }
+
+    BgSpot03Taki_ApplyOpeningAlpha(this, this->bufferIndex);
+}
+
+void BgSpot03Taki_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot03Taki* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void BgSpot03Taki_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot03Taki* this = THIS;
+    s32 pad;
+    u32 gameplayFrames;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_spot03_taki.c", 321);
+
+    gameplayFrames = globalCtx->gameplayFrames;
+
+    gSPMatrix(oGfxCtx->polyXlu.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_spot03_taki.c", 325),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    func_80093D84(globalCtx->state.gfxCtx);
+
+    gSPSegment(
+        oGfxCtx->polyXlu.p++, 0x08,
+        Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, gameplayFrames * 5, 64, 64, 1, 0, gameplayFrames * 5, 64, 64));
+
+    gSPDisplayList(oGfxCtx->polyXlu.p++, &D_06000B20);
```

Comment:
The `&` is omitted for arrays (and for function pointers). And this is relevant in the rest of the function.
```suggestion
    gSPDisplayList(oGfxCtx->polyXlu.p++, D_06000B20);
```

Reply:
Is this true for `DynaPolyInfo_Alloc` too? Almost all the uses of this function use `&`

Reply:
It's only true for arrays (eg. display lists are of type `Gfx[]`) but the argument of `DynaPolyInfo_Alloc` isn't an array so this doesn't apply.

Reply:
Right